src/language-js/print/call-arguments.js

```diff
@@ -0,0 +1,362 @@
+"use strict";
+
+const { printDanglingComments } = require("../../main/comments");
+const { getLast, getPenultimate } = require("../../common/util");
+const {
+  getFunctionParameters,
+  // [prettierx] --paren-spacing option support (...)
+  hasAddedLine,
+  hasComment,
+  CommentCheckFlags,
+  isFunctionCompositionArgs,
+  isJsxNode,
+  isLongCurriedCallExpression,
+  shouldPrintComma,
+  getCallArguments,
+  iterateCallArgumentsPath,
+  isNextLineEmpty,
+  isCallExpression,
+  isStringLiteral,
+  isObjectProperty,
+} = require("../utils");
+
+const {
+  builders: {
+    line,
+    hardline,
+    softline,
+    group,
+    indent,
+    conditionalGroup,
+    ifBreak,
+    breakParent,
+  },
+  utils: { willBreak },
+} = require("../../document");
+
+const { ArgExpansionBailout } = require("../../common/errors");
+const { isConciselyPrintedArray } = require("./array");
+
+function printCallArguments(path, options, print) {
+  const node = path.getValue();
+  const isDynamicImport = node.type === "ImportExpression";
+
+  // [prettierx] for --paren-spacing option support (...)
+  const parenSpace = options.spaceInParens ? " " : "";
+  const parenLine = options.spaceInParens ? line : softline;
+
+  const args = getCallArguments(node);
+  if (args.length === 0) {
+    return [
+      "(",
+      printDanglingComments(path, options, /* sameIndent */ true),
+      ")",
+    ];
+  }
+
+  // useEffect(() => { ... }, [foo, bar, baz])
+  if (isReactHookCallWithDepsArray(args)) {
+    // [prettierx] with spaceInParens option support (...)
+    return [
+      "(",
+      parenSpace,
+      print(["arguments", 0]),
+      ", ",
+      print(["arguments", 1]),
+      parenSpace,
+      ")",
+    ];
```

Comment:
It would be nice if we could use utility functions to help factor out patterns like this someday. It would be ideal if we could do this in the upstream Prettier project.